### DISCUSSION TITLE
How to enable by default agent ?

### DISCUSSION THREAD

**Discussion by Hraesvelg** [gocd/gocd#12966] [General]
Hello,

After up docker services (gocd-server and one gocd-agent) my agent is disabled by default, is it possible to activate automatically ? 

also, in env variables, i set :

cruise-config.xml

```
    ....
    <environments>
        <environment name="builder"/>
    </environments>
</cruise>
```

.env

`AGENT_AUTO_REGISTER_ENVIRONMENTS=builder`

but this setting seems have no effect !? how can i check if something has failed ?

Thanks.

**Comment by sriramnrn** [comment]
On Wed, 24 Jul 2024 at 4:21 AM, Freddy ***@***.***> wrote:

> Hello,
>
> After up docker services (gocd-server and one gocd-agent) my agent is
> disabled by default, is it possible to activate automatically ?
>
> also, in env variables, i set :
>
> cruise-config.xml
>
>     ....
>     <environments>
>         <environment name="builder"/>
>     </environments>
> </cruise>
>
> .env
>
> AGENT_AUTO_REGISTER_ENVIRONMENTS=builder
>
> but this setting seems have no effect !? how can i check if something has
> failed ?
>

Please see
https://docs.gocd.org/current/advanced_usage/agent_auto_register.html

**Comment by Hraesvelg** [comment]
hum ok i tested but without effect.. my file seems to be took in account but the final file in container :

cat /godata/config/autoregister.properties 

```
# Auto-registration properties in key=value format, encoded in UTF-8
agent.auto.register.key=a8cc1533-a366-4903-a71d-8cf017ae6f35
agent.auto.register.environments=builder
agent.auto.register.hostname=gocd-agent-1
# Auto-registration properties in key=value format, encoded in UTF-8
agent.auto.register.key=
agent.auto.register.resources=
agent.auto.register.environments=
agent.auto.register.hostname=
```

file seems overwrite

**Reply by chadlwilson** [reply]
When using the containers you have to use env vars available to the docker entry point. All documented - please read. https://hub.docker.com/r/gocd/gocd-agent-wolfi

Your original example probably isn't working because you didn't supply the most important thing - the key, or you didn't supply the env vars correctly to container startup.
